### DISCUSSION TITLE
rpmbuild/rpm not reading macros from the the rcfile

### DISCUSSION THREAD

**Discussion by rmehtacs** [rpm-software-management/rpm#2030] [Q&A] [answered]
I am learning rpm/rpmbuild and I see strange behavior difference in RPM v4.4.2.3 and RPM v4.14.
I created an **rpmrc** file which has below content

> macrofiles:/home/rmehta/rpmtest/rpmmacros

and **rpmmacros** has below contents

> %_name Rakesh

When I execute `rpmbuild --rcfile /home/rmehta/rpmtest/rpmrc --eval %_name` it gives different output for both RPM versions For 

> RPM version 4.4.2.3 (SLES-15) it prints **Rakesh**
> For RPM version 4.14.3 (SLES-11) it prints **%_name**

Also tried  ` --eval  '%{_name}'` and it has same result.

My question is why this difference and why for RPM v4.14.3 its not reading the rcfile.

I tried same with RPM v4.11(openSUSE) and its behaviour is same as RPM v4.14.3.

**Comment by pmatilai** [comment, marked as answer]
Support for `macrofiles:` directive in rpmrc was dropped in [rpm 4.6.0](https://rpm.org/wiki/Releases/4.6.0), 13 years ago. Rpm v4.4 is old and long obsolete, the only thing you'll learn from it is bad habits.

**Reply by rmehtacs** [reply]
@pmatilai  Thanks for answer. Can you share any alternatives for this? What's the new way to define own macros and overriding existing ones. and Thanks again for your answer. This will give me a new direction for solution.

**Comment by rmehtacs** [comment]
The problem was solved by copying rpmmacros file in home directory as .rpmmacros.

Modified makefiles and add below command

cp -f  <PATH_TO>/rpmmacros ~/.rpmmacros
